src/uu/cp/src/cp.rs

```diff
@@ -983,6 +983,16 @@ fn adjust_canonicalization(p: &Path) -> Cow<Path> {
     }
 }
 
+/// Decide whether the given path is a symbolic link.
+fn is_symlink(path: &Path) -> bool {
+    // TODO Replace this convenience function with `Path::is_symlink()`
+    // when the minimum supported version of Rust is 1.58 or greater.
+    match fs::symlink_metadata(path) {
+        Err(_) => false,
```

Comment:
Why not use the is_ok() method here instead?

Reply:
Because we need to call `file_type().is_symlink()` on the value stored in `Ok()`. Could you give an example of what you mean and why it would work better?